### ISSUE TITLE
Probe Whoa's 

### ISSUE THREAD

**Issue by synchromesh62** [TheGameCreators/GameGuruRepo#4376]
I know this is probably a low priority right now but that **Damn Probe.**
when it comes to darker interiors you start the level 90 % of the time much brighter in test, standalone and VR which just looks bad. Its not like it settles in after a second or two ( that wouldnt be so bad) but It will stay that way until you move around and it finally kicks in. The vid below shows the problem but looks a lot worse ingame and using VR

https://youtu.be/1waZ6Ffso54

**Comment by davetgc** [comment]
I think the lack of multiple probes has other issues too, for example reflections not catching up. Add to that the fact that Wicked is capable of creating very good looking scenes with multiple probes means it will become a bigger issue sooner rather than later in my opinion.

**Comment by synchromesh62** [comment]
> 

I think it should still be classed as a **bug** then to be honest. Not going to release a demo whilst its doing this.
Enhancement will just sit there unlikely to even be read by the team at this time. 
Not expecting anything done any time soon but an **Enhancement** ?

**Comment by davetgc** [comment]
Marked as both, as it could be considered a serious issue. I'll let the dev team decide.

**Comment by synchromesh62** [comment]
> Marked as both, as it could be considered a serious issue. I'll let the dev team decide.

Ye thats fine Dave thanks.
Its like a hospital appointment .. I wanted to at least be on the waiting list rather than we will get back to you :)

**Comment by davetgc** [comment]
lol. Very true.

**Comment by MonkeyFrogStudio** [comment]
The lack of user-placeable environment probes is a HUGE issue, imo. As another user pointed out, PBR relies on these probes to do what PBR does. It's vital. Else, we end up with something similar to Classic - something akin to PBR, but not quite there.

**Comment by davetgc** [comment]
I'm collocating a list of issues relating to this:

#4216

**Comment by MonkeyFrogStudio** [comment]
Some links:

https://docs.unrealengine.com/4.27/en-US/BuildingWorlds/LightingAndShadows/ReflectionEnvironment/

I will add more as I find 'em. :)

**Comment by bond1mjb** [comment]
The first video that @MonkeyFrogStudio posted about Wicked Engine has important information on the use of multiple probes at the 4:50 mark: https://youtu.be/2FXWesabBNE?t=264

**Comment by plemsoft** [comment]
We had some problems when we had many environment probe's , the current wicked version we use do not handle this correctly and we got crashes and strange visuals , i think this one should wait on the DX12 update, anyway its by design how it works now, i will assign this one to Lee to decide :)

**Comment by LeeBamberTGC** [comment]
@synchromesh62 New probe system in next EXP build, with added properties for lights in the advanced mode.  You can now opt in a light to be a light probe location and can set the range of the env map capture:
![image](https://user-images.githubusercontent.com/35840583/230136733-88675685-2c2b-4eef-8e23-0a594757dbd3.png)

Also added some nice smoothing stuff to transition calmly between the newly recruited 8 physical probes to cater for as many virtual probes as you need for your level

**Comment by synchromesh62** [comment]
Fantastic !!
Thanks Lee i will check it out :)

**Comment by MonkeyFrogStudio** [comment]
@LeeBamberTGC First of all, thank you so very much for adding these in so quickly! You don't know how much I appreciate it. I'm sure others do, too.

One thing that instantly stands out to me is we have no visual means of know the bounds of the light probe. Adjusting the probe Scale is not really cutting it. Most light/environment/reflection probes give the end-user a box that they can adjust to ensure that box totally encompasses the area it is meant to affect. With the current system, it's a bit of guesswork. 

I am guessing that the probe uses a cube? Especially since probes like these normally generate something similar to cube maps to work. If that's the case, we really need to be able to SEE that cube (just a wireframe outline) when we scale so we can ensure the probe convers the entire area. Maybe your probes (or Wicked's probes) work differently, though. In any case, I am finding it a bit difficult to adjust the probe by sort of guessing the scale.

Also, having some sort of visual indicator would help us to not overlap other probes used in the same level.

Next, why is this attached to a light? Probes should be independent of light sources, shouldn't they? Let's say I have a square room with a light in each corner. The probe would be placed in the center to work properly. But I don't want a light there. With the current system, I place a light, tick the Probe box, then I either set the light's range (not the probe) to zero and/or set the light color to black. It's not many steps and simple enough to do, but why not have a separate Probe Object? 

Lastly, what about control over a probe's resolution? Right now there is none. But most game engines let you choose any resolution from 128 to even 8k (though I wouldn't recommend anything above 2k).

Thanks again for adding this!

**Comment by MonkeyFrogStudio** [comment]
@LeeBamberTGC From the live broadcast - Probe resolution should be per probe and NOT global, please. Your reasons given there were correct. Thanks! :)

**Comment by LeeBamberTGC** [comment]
Moving the probe task to a single issue: https://github.com/TheGameCreators/GameGuruRepo/issues/3565#issuecomment-1498119571
